docs/viewlifecycle.md

```diff
@@ -0,0 +1,439 @@
+  * [View Creation Events](#view-creation-events)
+  * [View Destruction Events](#view-destruction-events-events)
+  * [Other View Events](#other-view-events)
+* [`CollectionView` Lifecycle](#collectionview-lifecycle)
+  * [Collection Creation Lifecycle](#collection-creation-lifecycle)
+  * [Collection Destruction Lifecycle](#collection-destruction-lifecycle)
+  * [Collection Creation Events](#collection-creation-events)
+  * [Collection Destruction Events](#collection-destruction-events-events)
+  * [Other Collection Events](#other-collection-events)
+
+## `View` Lifecycle
+Marionette views defined a number of events during the creation and destruction
+lifecycle - when the view is displayed in and emptied from a region. In the
+documentation, we will reference the event name, though `onEvent` handling can
+be used.
+
+### View Creation Lifecycle
+
+When a view is initialized and then displayed inside a region (using
+`showChildView()`) a set of events will be called in a specific order.
+
+| Order |      Event      |
+| :---: |-----------------|
+|   1   | `before:render` |
+|   2   | `render`        |
+|   3   | `before:attach` |
+|   4   | `attach`        |
+|   5   | `dom:refresh`   |
+
+### View Destruction Lifecycle
+
+When  `region.empty()` is called, the view will be destroyed, calling events as
+part of the destruction lifecycle.
+
+| Order |       Event       |
+| :---: |-------------------|
+|   1   |  `before:destroy` |
+|   2   |  `before:detach`  |
+|   3   |  `detach`         |
+|   4   |  `destroy`        |
+
+### View Creation Events
+
+These events are fired during the view's creation and rendering in a region.
+
+#### View `before:render`
+
+Triggered before a View is rendered.
+
+```javascript
+var Mn = require('backbone.marionette');
+
+Mn.View.extend({
+  onBeforeRender: function() {
+    // set up final bits just before rendering the view's `el`
+  }
+});
+```
+
+#### View `render`
+
+Triggered after the view has been rendered.
+You can implement this in your view to provide custom code for dealing
+with the view's `el` after it has been rendered.
+
+```javascript
+var Mn = require('backbone.marionette');
+
+Mn.View.extend({
+  onRender: function() {
+    console.log('el exists but is not visible in the DOM');
+  }
+});
+```
+
+#### View `before:attach`
+
+Triggered after the View has been rendered but just before it is first bound
+into the page DOM. This will only be triggered once per `region.show()` - if
+you want something that will be triggered every time the DOM changes,
+you may want `render` or `before:render`.
+
+#### View `attach`
+
+Triggered once the View has been bound into the DOM. This is only triggered
+once - the first time the View is attached to the DOM. If you want an event that
+triggers every time the DOM changes visibly, you may want `dom:refresh`
+
+#### View `dom:refresh`
+
+Triggered after the first `attach` event fires _and_ every time the visible DOM
```

Comment:
`dom:refresh` is sort of a combination of `attach` and `render`
I think I would go with saying

> Triggered after the first `attach` event fires _and_ the view's `render` is called while the view is in the DOM.

Although the explanation isn't that much more insightful. I think providing an example would help.

```
const myView = new Mn.View({
  template: _.template('<span><%= count %><span>'),
  templateContext: function(){
    this.count = (this.count || 0) + 1;
    return {
      count: this.count
    };
  }
});

myView.on('render', ()=> console.log('render'));
myView.on('attach', ()=> console.log('attach'));
myView.on('dom:refresh', ()=> console.log('dom:refresh'));

// made up variable
layout.showChildView('myRegion', myView);
// render, attach, and dom:refresh should fire

myView.render();
// render, and dom:refresh should fire